src/renderer/components/ProgramChangeView.tsx

```diff
@@ -1,92 +1,106 @@
-  MAX_PROGRAM_CHANGES_DISPLAYED,
-  ProgramAccountChange,
-  ProgramChangesState,
-  ProgramID,
-  WBAccount,
-} from '../../types/types';
+
+import { useAppSelector, useAppDispatch } from '../hooks';
+import { selectValidatorNetworkState } from '../data/ValidatorNetwork/validatornetworkSlice';
+import { BASE58_PUBKEY_REGEX, getAccount } from '../data/accounts/getAccount';
+import { AccountInfo } from '../data/accounts/accountsSlice';
+
 import Editable from './Editable';
-import ProgramChange from './ProgramChange';
+import { ProgramChange } from './ProgramChange';
+import createNewAccount from '../data/accounts/account';
+import {
+  unsubscribeProgramChanges,
+  subscribeProgramChanges,
+} from '../data/accounts/programChanges';
+import {
+  accountsActions,
+  selectAccountsListState,
+} from '../data/SelectedAccountsList/selectedaccountsSlice';
+
+export const MAX_PROGRAM_CHANGES_DISPLAYED = 20;
+export enum KnownProgramID {
+  SystemProgram = '11111111111111111111111111111111',
+  SerumDEXV3 = '9xQeWvG816bUx9EPjHmaT23yvVM2ZWbrrpZb9PusVFin',
+  TokenProgram = 'TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA',
+}
+
+interface PinnedAccountMap {
+  [pubKey: string]: boolean;
+}
 
 function ProgramChangeView(props: {
-  accounts: WBAccount[];
   attemptAccountAdd: (pubKey: string, initializing: boolean) => void;
 }) {
-  const dispatch = useDispatch();
-  const { accounts, attemptAccountAdd } = props;
-  const { net } = useSelector((state: RootState) => state.validator);
+  const dispatch = useAppDispatch();
+  const { attemptAccountAdd } = props;
+  const { net } = useAppSelector(selectValidatorNetworkState);
+
+  // TODO: I suspect It would be nicer to use a function need to try it..
+  const selectAccounts = useAppSelector(selectAccountsListState);
+  const { pinnedAccounts } = selectAccounts;
+
+  const pinAccount = (pubKey: string, pinned: boolean) => {
+    if (!pinned) {
+      getAccount(net, pubKey)
+        .then((res) => {
+          // eslint-disable-next-line promise/always-return
+          if (res) {
+            if (res instanceof Error) {
+              return;
+            }
+            dispatch(accountsActions.unshift(res));
+          }
+        })
+        /* eslint-disable no-console */
+        .catch(console.log);
+    } else {
+      dispatch(accountsActions.rm(pubKey));
+    }
+  };
 
   // want to check paused before updating changes later,
   // so we include these together
-  const [changesState, setChangesState] = useState<ProgramChangesState>({
-    changes: [],
-    paused: false,
+  const [changes, setChangesState] = useState<AccountInfo[]>([]);
+  const [paused, setPausedState] = useState<boolean>(false);
+
+  const displayList: string[] = []; // list of Keys
+  const pinnedAccount: PinnedAccountMap = {};
+
+  // this should be used to build the new list for display, not to mutate the changes list...
+  // eslint-disable-next-line array-callback-return
+  pinnedAccounts.map((key: string): void => {
```

Comment:
this should really be like a `forEach` or something right? `map` almost always is expected to return something

Reply:
yeah, this one has been confusing me - dunno why brb